src/ome_zarr_models/_rfc5_transforms/multiscales.py

```diff
@@ -0,0 +1,247 @@
+                transform.transformations,
+                valid_lengths=[2],
+                variable_name="transform.transforms (i.e. transformations composing "
+                "the sequence)",
+            )
+            first, second = transform.transformations
+            if first.type != "scale":
+                msg = (
+                    "When the first (and only) element in `coordinateTransformations`"
+                    " is a `Sequence`, the first element must be a `Scale` transform. "
+                    f"Got {first} instead."
+                )
+                raise ValueError(msg)
+            if second.type != "translation":
+                msg = (
+                    "When the first (and only) element in `coordinateTransformations`"
+                    " is a `Sequence`, the second element must be a `Translation` "
+                    f"transform. Got {second} instead."
+                )
+                raise ValueError(msg)
+        elif transform.type != "scale":
+            msg = (
+                "The first transformation in `coordinateTransformations` "
+                "must either be a `Scale` transform or a `Sequence` transform. "
+                f"Got {transform} instead."
+            )
+            raise ValueError(msg)
+
+        return transforms_obj
+
+    @field_validator("coordinateTransformations", mode="after")
+    @classmethod
+    def _ensure_transform_dimensionality(
+        cls,
+        transforms: tuple[CoordinateTransformationType, ...],
+    ) -> tuple[CoordinateTransformationType, ...]:
+        """
+        Ensures that the dimensionality of the scale and translation (when both present)
+        match
+        """
+        maybe_sequence = transforms[0]
+        if maybe_sequence.type == "sequence":
+            first, second = maybe_sequence.transformations
+            if (
+                isinstance(first, Scale)
+                and isinstance(second, Translation)
+                and len(first.scale) != len(second.translation)
+            ):
+                raise ValueError(
+                    "The length of the scale and translation vectors must be the same."
+                    f"Got {len(first.scale)} and {len(second.translation)}."
+                )
+        return transforms
+
+
+class Multiscale(BaseAttrs):
+    """
+    An element of multiscales metadata.
+    """
+
+    coordinateSystems: tuple[CoordinateSystem, ...] = Field(..., min_length=1)
+    datasets: tuple[Dataset, ...] = Field(..., min_length=1)
+    coordinateTransformations: tuple[CoordinateTransformationType, ...] | None = None
+    metadata: JsonValue = None
+    name: JsonValue | None = None
+    type: JsonValue = None
+
+    @property
+    def ndim(self) -> int:
+        """
+        Dimensionality of the data described by this metadata.
+
+        Determined by the length of the axes attribute.
+        """
+        output_cs_name = self.datasets[0].coordinateTransformations[0].output
```

Comment:
Instead of taking the first dataset here, shouldn't `ndim` be a property on `Dataset` instead (because different datasets in multiscales could have different dimensionalities)?

Reply:
In the old version of transformations all datasets had the same dimensionality (this was validated here https://github.com/ome-zarr-models/ome-zarr-models-py/blob/0f7fb161b4bd8d34e1b7106cbb5fcfc6c50910c0/src/ome_zarr_models/common/multiscales.py#L169)

Now in the new specs (see also https://github.com/ome-zarr-models/ome-zarr-models-py/pull/187#discussion_r2087541247), has this changed? As in the linked comment, is this what the new specs want?

Reply:
I'm re-reading the spec as altered by RFC 5, and I think it implies (but doesn't explicitly say) that all datasets have the same dimensionality, so we can assume that. So I think the code as written is valid/fine.

It might be a bit simpler to get `ndim` from `self.coordinateSystems[0]` here though?